### PR TITLE
amd-power-control: modify power restore policy

### PR DESCRIPTION
**Pull request by modolaty**
modify power restore policy for checking the
proper value and add debug log

Signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>

### CODE REVIEW

File: power-control-x86/src/power_control.cpp [hunk 1]
```diff
@@ -614,50 +614,48 @@ static void powerRestorePolicyCheck()
...
         const std::string* policy =
                 std::get_if<std::string>(&restorepolicy);
         if (policy == nullptr)
         {
-            std::cerr << "Unable to read restore policy\n";
+            std::cerr << "Unable to read power restore policy\n";
             return;
         }
 
-        if (policy->compare("None") == 0)
```
Comment (supven01) [inline power-control-x86/src/power_control.cpp:627]: We still need this statement.

File: power-control-x86/src/power_control.cpp [hunk 2]
```diff
@@ -614,50 +614,48 @@ static void powerRestorePolicyCheck()
...
-        else
+        else if (policy->compare("xyz.openbmc_project.Control.Power.RestorePolicy.Policy.Restore") == 0)
         {
-            std::cerr << "Invoking Restore Policy: " << policy << "\n";
-
-            sd_journal_send("MESSAGE=PowerControl: power restore policy applied",
-                    "PRIORITY=%i", LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.PowerRestorePolicyApplied", NULL);
-
-            if (policy->compare("AlwaysOn") == 0)
```
Comment (supven01) [inline power-control-x86/src/power_control.cpp:639]: Change this condition.

File: power-control-x86/src/power_control.cpp [hunk 3]
```diff
@@ -614,50 +614,48 @@ static void powerRestorePolicyCheck()
...
+            if (wasPowerDropped())
             {
+                std::cerr << "Power was dropped, restoring Host to On state\n";
                 sendPowerControlEvent(Event::powerOnRequest);
-                setRestartCauseProperty(getRestartCause(RestartCause::powerPolicyOn));
+                setRestartCauseProperty(getRestartCause(RestartCause::powerPolicyRestore));
             }
-            else if (policy->compare("Policy.Restore") == 0)
```
Comment (supven01) [inline power-control-x86/src/power_control.cpp:644]: Change this condition.

File: power-control-x86/src/power_control.cpp (hunk 1 above)
Comment (supven01) [reply power-control-x86/src/power_control.cpp:627]: However, the condition itself may need to initialized to proper value.

File: power-control-x86/src/power_control.cpp [hunk 4]
```diff
@@ -614,50 +614,48 @@ static void powerRestorePolicyCheck()
...
-            std::cerr << "Invoking Restore Policy: " << policy << "\n";
-
-            sd_journal_send("MESSAGE=PowerControl: power restore policy applied",
-                    "PRIORITY=%i", LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.PowerRestorePolicyApplied", NULL);
-
-            if (policy->compare("AlwaysOn") == 0)
+            if (wasPowerDropped())
             {
+                std::cerr << "Power was dropped, restoring Host to On state\n";
```
Comment (supven01) [inline power-control-x86/src/power_control.cpp:640]: extra print statements ok, but please don't remove if else statements.

**Review by supven01 — CHANGES_REQUESTED** [review]

File: power-control-x86/src/power_control.cpp (hunk 1 above)
Comment (modolaty) [reply power-control-x86/src/power_control.cpp:627]: It is now moved to the end of if else statement as the last else to catch all other conditions

        else if (policy->compare("xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOff") == 0)
            std::cerr << "Restore policy to Always Off\n";
        else
            std::cerr << "Restore policy to None\n";

File: power-control-x86/src/power_control.cpp (hunk 2 above)
Comment (modolaty) [reply power-control-x86/src/power_control.cpp:639]: it is done with:
if (policy->compare("xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOn") == 0)

File: power-control-x86/src/power_control.cpp (hunk 3 above)
Comment (modolaty) [reply power-control-x86/src/power_control.cpp:644]: Done with 
else if (policy->compare("xyz.openbmc_project.Control.Power.RestorePolicy.Policy.Restore") == 0)

**Pull request closed without merging**